Fix/sonarqube

jwsong98 opened the pull request:
1. 사용하지 않은 Event 패키지 제거
 원래 email 보내는 작업의 성공 여부와 상관없이, 도메인 작업의 결과를 응답하기 위해, 
 반응 속도 측면도 고려하여 event-driven 방식으로 email을 보내도록 했지만 사용하지 않아서 삭제

2. DAO 패키지 명 수정 -> dao

3. event가 reservation을 이동시킬 때,  자기가 가지고 있는 모든 reservation을 지웠는데, 의도한 대로 필터가 안된 것들만 삭제되게 수정

**4. Builder에서 기본 속성으로 new ArrayList()를 호출하는 것이 무시되던 사항을 final을 붙여 수정 (-> event.addOneReseravtion 오류의 원인)**

5. converter들 private 생성자 추가

6. admin/repository/tempReservationRepository 삭제

**7. Controller에서 ResponeEntity의 Generic type을 지정하지 않은 부분 전부 추가**

**8. optional에서 empty인 것을 확인하지 않고 가져온 것, 확인하고 가져오게 수정 ReservationService / 
validateMaxReservation**

9. 리스트 안에 객체들을 순회하며 조건에 만족하는 bookmark를 list.remove를 이용해 삭제하던 걸 iterator를 이용해 삭제

10. Security Config에서 필터링 하는 문자열 순서를 빡빡하게-> 느슨하게 조건을 거는 순서로 수정

jwsong98 committed:
fix : 시간대 체크 하는 함수 변경

jwsong98 committed:
fix : 소나큐브 critical, major severity 해결

jwsong98 committed:
fix : remove test

jwsong98 committed:
fix : .orElseThrow 안하고 get 한 거 수정

jwsong98 committed:
fix : api 수정

shortboy7 commented:
API 수정
GET /api/reservation/admin/all ->  /api/admin/reservation/all
GET /api/reservation/admin/{userId} -> /api/admin/reservation/{userId}
GET /api/reservation/admin/approval/{reservationId} -> /api/admin/reservation/approval/{reservationId}
GET /api/reservation/amin/reject/{reservationId} -> /api/admin/reservation/reject/{reservationId}

POST /master/users/empowerment -> /api/master/users/empowerment
POST /master/users/deprivation -> /api/master/users/deprivation
POST /master/users -> /api/master/users

이렇게 수정하였고, domain 구조도 변경함

pletain committed:
fix: Severity 문제 해결
같은 클래스 내에서 @Transactional 중첩 발생 문제 해결
예약 정보 변경 체크 메소드에서 LocalTime과 LocalDate 비교 오류 정정

pletain commented:
수정해주신 내역 확인 완료했습니다.
